### PR TITLE
Enable attachments

### PR DESCRIPTION
**Pull request by victorzheng02**

### CODE REVIEW

File: backend/routes/files.routes.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    '/bulk/:reference',
+    authenticateUser,
+    upload.array('files'),
+    FilesController.bulkCreateFileController
+)
+router.delete(
+    '/bulk',
+    authenticateUser,
+    FilesController.bulkDeleteFileController
+)
```
Comment (ansonjwhe) [inline backend/routes/files.routes.js:30]: why is the pattern for this different from the bulk post? if we're not using this yet, we can remove it now and add it later.

File: backend/controller/files.controller.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+const bulkCreateFileController = (req, res) => {
+    bulkCreateFiles(req.files, req.params.reference)
+        .then((newFiles) => res.status(200).json(newFiles))
+        .catch((err) => res.status(500).json('Error: ' + err))
+}
+
+const bulkDeleteFileController = (req, res) => {
+    const { ids } = req.body
+    const promises = ids.map((id) => deleteFile(id))
+    Promise.all(promises)
```
Comment (ansonjwhe) [inline backend/controller/files.controller.js]: lets create a `bulkDeleteFiles` to match the pattern of bulk create

File: backend/service/files.service.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+const File = require('../models/files.model')
+
+const getFile = async (id) => {
+    return File.findById(id)
+}
+
+const getAllFilesByReference = async (reference) => {
+    return File.find({ referenceItem: reference })
```
Comment (ansonjwhe) [inline backend/service/files.service.js:8]: can we rename "reference" to ticketId to be explicit about what is being referenced

File: backend/routes/files.routes.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+const FilesController = require('../controller/files.controller')
+const { authenticateUser } = require('../auth/middleware')
+const multer = require('multer')
+const storage = multer.memoryStorage()
+const upload = multer({ storage: storage })
+router.get('/:id', authenticateUser, FilesController.getFileController)
+router.get(
+    '/getallbyreference/:reference',
+    authenticateUser,
+    FilesController.getAllFilesByReferenceController
```
Comment (ansonjwhe) [inline backend/routes/files.routes.js:11]: rename "reference" to "ticketid"

File: backend/models/files.model.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
+const mongoose = require('mongoose')
+
+const { Schema } = mongoose
+
+const FileSchema = new Schema(
+    {
+        // reference a file by its id
+        referenceItem: { type: Number, index: true },
```
Comment (ansonjwhe) [inline backend/models/files.model.js:8]: rename to `ticketId`

File: frontend/src/pages/Dashboard.js [hunk 6]
```diff
@@ -293,6 +306,14 @@ const Dashboard = () => {
                 <TicketList isLoading={isLoading} />
                 {getMainContent()}
             </Flex>
+            {isUploadModalOpen && (
+                <UploadFileModal
+                    isOpen={isUploadModalOpen}
+                    onClose={onCloseUploadModal}
+                    endpointToSave={TICKET_ENDPOINTS.UPR}
```
Comment (ansonjwhe) [inline frontend/src/pages/Dashboard.js]: remove because unused

**Review by ansonjwhe — CHANGES_REQUESTED** [review]

File: backend/service/files.service.js (hunk 3 above)
Comment (victorzheng02) [reply backend/service/files.service.js:8]: This and subsequent comments resolved via call. Naming it reference as we wish to remain generic, and things might be tied to non-ticket concepts such as users

**Review by ansonjwhe — APPROVED** [review]

**Pull request merged**